contracts/test/L1ETHRegistry.t.sol

```diff
@@ -10,110 +10,118 @@ import "../src/L1/L1ETHRegistry.sol";
 import "../src/common/RegistryDatastore.sol";
 import "../src/common/IRegistry.sol";
 import "../src/L1/IL1EjectionController.sol";
-import "../src/common/EnhancedAccessControl.sol";
+import {EnhancedAccessControl} from "../src/common/EnhancedAccessControl.sol";
 import "../src/common/IRegistryMetadata.sol";
+import {RegistryRolesMixin} from "../src/common/RegistryRolesMixin.sol";
+import "../src/common/BaseRegistry.sol";
+import "../src/common/IStandardRegistry.sol";
+import "../src/common/ETHRegistry.sol";
 
 contract MockRegistryMetadata is IRegistryMetadata {
     function tokenUri(uint256) external pure override returns (string memory) {
         return "";
     }
 }
 
-contract TestL1ETHRegistry is Test, ERC1155Holder {
-    event TransferSingle(address indexed operator, address indexed from, address indexed to, uint256 id, uint256 value);
-
+contract TestL1ETHRegistry is Test, ERC1155Holder, RegistryRolesMixin, EnhancedAccessControl {
     RegistryDatastore datastore;
     L1ETHRegistry registry;
     MockEjectionController ejectionController;
     MockRegistryMetadata registryMetadata;
+    address constant MOCK_RESOLVER = address(0xabcd);
 
     uint256 labelHash = uint256(keccak256("test"));
 
+    function supportsInterface(bytes4 /*interfaceId*/) public pure override(ERC1155Holder, EnhancedAccessControl) returns (bool) {
+        return true;
+    }
+    
     function setUp() public {
         datastore = new RegistryDatastore();
         ejectionController = new MockEjectionController();
         registryMetadata = new MockRegistryMetadata();
-        registry = new L1ETHRegistry(datastore, address(ejectionController), registryMetadata);
+        registry = new L1ETHRegistry(datastore, registryMetadata, ejectionController);
     }
 
     function test_eject_from_namechain_unlocked() public {
         vm.prank(address(ejectionController));
-        uint256 tokenId = registry.ejectFromNamechain(labelHash, address(this), registry, uint64(block.timestamp) + 86400);
+        uint256 tokenId = registry.ejectFromNamechain(labelHash, address(this), IRegistry(address(registry)), MOCK_RESOLVER, uint64(block.timestamp) + 86400);
```

Comment:
same redundant conversion applies here and 8-9 other lines in the same file as well for `IRegistry(address(registry))` since `L1ETHRegistry` implements `IRegistry` in the base.